### ISSUE TITLE
TextField will never get focus again once after above route is poped by user gesture on iOS

### ISSUE THREAD

**Issue by fujidaiti** [flutter/flutter#149478]
Suppose we have a textfield in screen A, and screen B stacked on top of A. Once B is poped by a swipe gesture, the textfield will never get focus again. And after some interactions with the textfield, I got the following exception. The full log is available in Logs section.
```console
A ValueNotifier<MagnifierInfo> was used after being disposed.
```

### Steps to reproduce

1. Run the code below on iOS.
2. Tap the `Go to B` button.
3. Pop screen B by swiping.
4. Tap the textfield in screen A.

### Expected results

The textfield can get focus even after B is popped by a user gesture.

### Actual results

The textfield will never get focus after B is popped by a user gesture.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const TextFieldFocusProblem());
}

class TextFieldFocusProblem extends StatelessWidget {
  const TextFieldFocusProblem({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: ScreenA(),
    );
  }
}

class ScreenA extends StatelessWidget {
  const ScreenA({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            const TextField(),
            TextButton(
              onPressed: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(
                    builder: (_) => const ScreenB(),
                  ),
                );
              },
              child: const Text('Go to B'),
            ),
          ],
        ),
      ),
    );
  }
}

class ScreenB extends StatelessWidget {
  const ScreenB({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: const Center(
        child: Text('B'),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/68946713/4eea8085-1dfb-41e6-8dd3-309af2220990

</details>

### Logs

<details open><summary>Logs</summary>

```console
Restarted application in 251ms.

════════ Exception caught by gesture ═══════════════════════════════════════════
The following assertion was thrown while handling a gesture:
A ValueNotifier<MagnifierInfo> was used after being disposed.
Once you have called dispose() on a ValueNotifier<MagnifierInfo>, it can no longer be used.

When the exception was thrown, this was the stack:
#0      ChangeNotifier.debugAssertNotDisposed.<anonymous closure> (package:flutter/src/foundation/change_notifier.dart:179:9)
#1      ChangeNotifier.debugAssertNotDisposed (package:flutter/src/foundation/change_notifier.dart:186:6)
#2      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:412:27)
#3      ValueNotifier.value= (package:flutter/src/foundation/change_notifier.dart:555:5)
#4      SelectionOverlay.updateMagnifier (package:flutter/src/widgets/text_selection.dart:1664:20)
#5      TextSelectionOverlay._handleSelectionEndHandleDragUpdate (package:flutter/src/widgets/text_selection.dart:779:24)
#6      SelectionOverlay._handleEndHandleDragUpdate (package:flutter/src/widgets/text_selection.dart:1211:28)
#7      DragGestureRecognizer._checkUpdate.<anonymous closure> (package:flutter/src/gestures/monodrag.dart:806:55)
#8      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:344:24)
#9      DragGestureRecognizer._checkUpdate (package:flutter/src/gestures/monodrag.dart:806:7)
#10     DragGestureRecognizer.handleEvent (package:flutter/src/gestures/monodrag.dart:652:11)
#11     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:98:12)
#12     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:143:9)
#13     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:633:13)
#14     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:141:18)
#15     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:127:7)
#16     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:495:19)
#17     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:475:22)
#18     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:425:11)
#19     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:420:7)
#20     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:383:5)
#21     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:330:7)
#22     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:299:9)
#23     _invoke1 (dart:ui/hooks.dart:328:13)
#24     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:442:7)
#25     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)

Handler: "onUpdate"
Recognizer: PanGestureRecognizer#ccb53
    debugOwner: _SelectionHandleOverlayState#81c0b(ticker inactive)
    start behavior: start
════════════════════════════════════════════════════════════════════════════════
```
</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
fvm flutter doctor -v
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.5 23F79 darwin-arm64, locale ja-JP)
    • Flutter version 3.22.0 on channel stable at /Users/fujidaiti/.fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (3 weeks ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/fujidaiti/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.87.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.90.0

[✓] Connected device (4 available)
    • iPhone 15 (mobile)              • 560C03AD-491E-41EC-AFDE-1A45E61E45C0 • iOS            •
      com.apple.CoreSimulator.SimRuntime.iOS-17-4 (simulator)
    • macOS (desktop)                 • macos                                • darwin-arm64   •
      macOS 14.5 23F79 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         •
      macOS 14.5 23F79 darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript •
      Google Chrome 125.0.6422.114

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by ebwood** [comment]
The related issue [148867](https://github.com/flutter/flutter/issues/148867)

**Comment by JosephNK** [comment]
I think this issue is critical, but I don't know why there hasn't been a hotfix yet. :(

**Comment by PhantomRay** [comment]
definitely should be a critical issue

**Comment by darshankawar** [comment]
Closing as duplicate of https://github.com/flutter/flutter/issues/148867.

I think a hotfix has been processed and most probably will be available soon : https://github.com/flutter/flutter/pull/149378

**Comment by ismailcaakir** [comment]
Unfortunately I solved the problem by downgrading to 3.19.0 on the stable channel.
